cpp/src/parquet/encoding.cc

```diff
@@ -2181,12 +2181,14 @@ class DeltaBitPackDecoder : public DecoderImpl, virtual public TypedDecoder<DTyp
   }
 
   int GetInternal(T* buffer, int max_values) {
-    max_values = std::min(max_values, this->num_values_);
+    int total_value_count = static_cast<int>(std::min(
+        total_value_count_, static_cast<uint32_t>(std::numeric_limits<int>::max())));
+    max_values = std::min(max_values, total_value_count);
```

Comment:
This is extremely convoluted. How about:
```suggestion
    max_values = static_cast<int>(std::min<int64_t>(max_values, total_value_count_));
```